platform/transcript.go

```diff
@@ -27,8 +27,39 @@ import (
 	"github.com/sashabaranov/go-openai"
 )
 
-// The total segments in overlay HLS.
-const maxOverlaySegments = 9
+// The default total segments in overlay HLS.
+const defaultMaxOverlaySegments = 9
+
+// Get the total segments in overlay HLS.
+func GetMaxOverlaySegments() (int, error) {
```

Comment:
The function `GetMaxOverlaySegments` should be designed to ignore any error:

```go
func GetMaxOverlaySegments() (int, error)
```

Use it in the transcript while ignoring any error:

```go
maxOverlaySegments, _ := GetMaxOverlaySegments()
```

In the `doMain`, after `setEnvDefault`, check for the error:

```go
setEnvDefault("SRS_TRANSCRIPT_OVERLAY_QUEUE_LIMIT", "9")
if _, err := GetMaxOverlaySegments(); err != nil {
    return errors.Wrapf(......)
```

If the environment is not set as expected, it should fail in `main`, not in the transcript.